frame/assets/src/mock.rs

```diff
@@ -89,6 +89,7 @@ impl Config for Test {
 	type Balance = u64;
 	type AssetId = u32;
 	type Currency = Balances;
+	type CreateOrigin = AsEnsureOriginWithArg<frame_system::EnsureSigned<u64>>;
```

Comment:
Should this really be the asset id rather than the balance?
```suggestion
	type CreateOrigin = AsEnsureOriginWithArg<frame_system::EnsureSigned<u32>>;
```

Reply:
Neither, it's `AccountId`
https://github.com/paritytech/substrate/pull/12586/files#diff-ff4e659ba36996f5041551dc830e12981f6c9e3074be9547063e943e3bf1f2c7R58

Reply:
in lib.rs's create there's the arg `id: T::AssetId,` that we then pass to the CreateOrigin:
			`let owner = T::CreateOrigin::ensure_origin(origin, &id)?;`
That was what was confusing me. Is it sometimes AssetId and sometimes AccountId dependent on context?

Reply:
In `fn ensure_origin` yes, but the `AsEnsureOriginWithArg` impl doesn't use the second argument: https://github.com/paritytech/substrate/blob/joe-assets-create-origin/frame/support/src/traits/dispatch.rs#L109-L133